src/test/java/io/github/oliviercailloux/y2018/apartments/apartment/ApartmentFactoryTest.java

```diff
@@ -66,13 +66,29 @@ void testGenerateApartmentFromParameters() {
 	}
 
 	/**
-	 * This function allow us to test if we have an exception thrown when the json path given
-	 * in argument of the method is wrong.
+	 * This function allow us to test if we have an exception thrown when the json
+	 * path given in argument of the method is wrong.
 	 */
 	@Test
 	public void generateApartmentFromJsonExceptionTest() {
-		assertThrows(IOException.class,
-				()-> ApartmentFactory.generateApartmentFromJsonPath(Path.of("abc")));
+		assertThrows(IOException.class, () -> ApartmentFactory.generateApartmentsFromJsonPath(Path.of("abc")));
+	}
+
+	/**
+	 * Make an API call and verify that the return format of the API has not changed
+	 */
+	@Test
+	public void testRandomAddress() {
```

Comment:
Il est inadéquat de dupliquer la logique de parsing dans ce test. Il devrait plutôt tester votre code que ses hypothèses. Il faut plutôt appeler une méthode de votre classe en lui passant cette lat, long et en vérifiant le résultat obtenu (méthode à créer si nécessaire, visibilité package). Cela aura également pour effet que le test détectera si le format de retour a changé (ce qu’on veut effectivement), mais tout en liant le test et l’implémentation réelle.

Reply:
Fix in 9f943e7e76d08a432b94484f61720a14c64e7ffd